bin/node/src/cli.rs

```diff
@@ -14,121 +9,54 @@ pub struct Cli {
-                                    .map_err(|_| Error::KeyStoreOperation)
-                            },
-                            Ok,
-                        )?;
-                        auth_keys
-                            .get_mut(&key_type)
-                            .unwrap()
-                            .push(*key.as_array_ref());
-                    }
-                    aleph_primitives::KEY_TYPE => {
-                        let keys = SyncCryptoStore::ed25519_public_keys(&*keystore, key_type);
-                        let key = keys.into_iter().next().map_or_else(
-                            || {
-                                SyncCryptoStore::ed25519_generate_new(&*keystore, key_type, None)
-                                    .map_err(|_| Error::KeyStoreOperation)
-                            },
-                            Ok,
-                        )?;
-                        auth_keys
-                            .get_mut(&key_type)
-                            .unwrap()
-                            .push(*key.as_array_ref());
-                    }
-                    _ => return Err(Error::Input("Unsupported key type".into())),
-                }
-            }
-        }
-
-        let keys_path = crate::chain_spec::KEY_PATH;
-        let auth_keys: HashMap<_, _> = auth_keys.iter().map(|(k, v)| (u32::from(**k), v)).collect();
-        let auth_keys = serde_json::to_string(&auth_keys).map_err(|e| Error::Io(e.into()))?;
-        std::fs::write(keys_path, &auth_keys).map_err(Error::Io)?;
-
-        Ok(())
+    fn load_spec(&self, path: &str) -> Result<Box<dyn sc_service::ChainSpec>, String> {
+        Ok(Box::new(chain_spec::ChainSpec::from_json_file(
+            std::path::PathBuf::from(path),
+        )?))
     }
 
-    fn open_keystore(&self, authority: &str) -> Result<SyncCryptoStorePtr, Error> {
-        let base_path: BasePath = self
-            .shared_params
-            .base_path()
-            .unwrap()
-            .path()
-            .join(authority)
-            .into();
-        let chain_id = self.shared_params.chain_id(self.shared_params.is_dev());
-        let config_dir = base_path.config_dir(&chain_id);
-
-        match self.keystore_params.keystore_config(&config_dir)? {
-            (_, KeystoreConfig::Path { path, password }) => {
-                Ok(Arc::new(LocalKeystore::open(path, password)?))
-            }
-            _ => unreachable!("keystore_config always returns path and password; qed"),
-        }
+    fn native_runtime_version(_: &Box<dyn ChainSpec>) -> &'static RuntimeVersion {
+        &aleph_runtime::VERSION
     }
 }
 
 #[derive(Debug, StructOpt)]
 pub enum Subcommand {
     /// Key management cli utilities
     Key(sc_cli::KeySubcommand),
-    /// Build a chain specification.
-    BuildSpec(sc_cli::BuildSpecCmd),
+
+    // NOTE: similarly we could have a BootstrapNode command that takes a node-name parameter
```

Comment:
Instead of a note in the code, it would be better to have a task in jira.

Reply:
I wasn't sure we want to do it in the end, but I wanted it mentioned somewhere (rather than nowhere :) ) Anyway: https://cardinal-cryptography.atlassian.net/browse/A0-368